src/adapter-manager.js

```diff
@@ -348,7 +350,7 @@ class AdapterManager extends EventEmitter {
     const packageJson = path.join(adapterPath, 'package.json');
     if (!fs.lstatSync(packageJson).isFile()) {
       console.error('package.json not found for adapter:', adapterId);
-      return;
+      return Promise.reject();
```

Comment:
All uses of `Promise.reject` should provide an error message explaining where the failure occurred. Since this is an async function, you can also use `throw new Error('blah')` to get a native error object and all that nice business